ddtrace/internal/datastreams/processor.py

```diff
@@ -138,7 +138,8 @@ def __init__(self, agent_url, interval=None, timeout=1.0, retry_attempts=3):
             initial_wait=0.618 * self.interval / (1.618 ** retry_attempts) / 2,
         )(self._flush_stats)
 
-        register_on_exit_signal(partial(_atexit, obj=self))
+        if threading.current_thread() is threading.main_thread():
```

Comment:
This assertion will fail for python2.7. This is fine for 2.x but we will need to update this comparison if we want to backport this fix to 1.20.

Reply:
yup; I will change the code to handle python2.7 as part of the backport